plugins/dell/meson.build

```diff
@@ -1,4 +1,11 @@
-if get_option('plugin_dell')
+tpm2tss_dell = dependency('tss2-esys', version : '>= 2.0', required: get_option('plugin_dell'))
+libsmbios_c = dependency('libsmbios_c', version : '>= 2.4.0', required: get_option('plugin_dell'))
+
+if tpm2tss_dell.found() and libsmbios_c.found()
+  if not get_option('plugin_uefi_capsule')
+    error('plugin_dell also needs plugin_uefi_capsule to work')
+  endif
```

Comment:
This check seems a bit reversed. If plugin_dell=auto, then the feature is optional. If the optional feature's *dependencies* are autodetected, but plugin_uefi_capsule which it needs is not auto-selected, you want to disable the optional feature.

If plugin_dell=enabled then you have *three* requirements:
- find tss2-esys
- find libsmbios_c
- select the `plugin_uefi_capsule` option

So what you really want here is:

```meson
if get_option('plugin_dell').enabled() and not get_option('plugin_uefi_capsule')
    error('plugin_dell is force-enabled but plugin_uefi_capsule is false, these options are incompatible')
elif tpm2tss_dell.found() and libsmbios_c.found()
    # optional, requirements met. Continue processing this file and build relevant targets
```